pulumi/infra/ec2.py

```diff
@@ -40,7 +40,7 @@ def allow_internally(
 
     def allow_outbound_any_ip(self, sg: aws.ec2.SecurityGroup) -> None:
         aws.ec2.SecurityGroupRule(
-            f"outbound-any-ip-egress-{self}",
+            f"outbound-any-ip-egress-{self}-to-{sg._name}",
```

Comment:
Can't comment on the actual code because it's outside Github's commenting window, but I think what is currently the `__str__` implementation for `Ec2Port` is closer to (but not quite) a `__repr__` implementation. I think having a `__str__` of

``` python
def __str__(self) -> str:
    return f"{self.protocol}:{self.port}"
```
would yield a more readable (and shorter) Pulumi resource name. 

Compare:
```
outbound-any-ip-egress-tcp:443-to-my-security-group
```
with
```
outbound-any-ip-egress-Ec2Port(tcp:443)-to-my-security-group
```